xmtp_id/src/associations/hashes.rs

```diff
@@ -0,0 +1,12 @@
+use sha2::{Digest, Sha256};
+
+pub fn sha256_string(input: String) -> String {
```

Comment:
What's the benefit to making this a string vs passing around some bytes?

Reply:
We could maybe make a new type, `Xid` that wraps a `Vec<u8>` and has a `Display` impl that outputs the `String` impl. This is probably a tad more efficient, but mostly it would be more obvious in-code when we have an XID vs just a String

Reply:
This is only used for the `xid` right now. I'd like the `xid` to be a string, since it makes it easier to handle at the platform SDK level. And there isn't much value in keeping it as bytes, since it doesn't represent a public key or anything like that.

Reply:
That makes sense

Reply:
I'd be OK with a type called XID that wraps a string to get that clarity. I just know that we're already annoying the platform SDK teams having `installation_id` be bytes. That one has a good reason (it's a public key), but this one is harder to justify.